client/me/purchases/manage-purchase/index.jsx

```diff
@@ -186,6 +189,10 @@ const ManagePurchase = React.createClass( {
 		const purchase = getPurchase( this.props );
 		let text;
 
+		if ( 'credit-card-updated' === this.props.destinationType ) {
+			text = this.translate( 'Your credit card details were successfully updated.' );
```

Comment:
One of the things I like about the other approach is that the success message comes from the server. We could keep the old approach for the notice, and just use the destination to see if the data changes.